functions/src/definitions/userHandlers.ts

```diff
@@ -509,7 +545,7 @@ async function newImageInstanceHandler({
     }
   }
 
-  if (!hasMatch) {
+  if (!hasMatch || !matchedInstanceSnap) {
     let writeResult = await db.collection("messages").add({
```

Comment:
I think the one you changed to (!hasMatch || !matchedInstanceSnap) won't work cos there are cases where hasMatched is true but matchedInstanceSnap is not assigned, and matchedParentMessageRef is assigned instead

Reply:
in that case, the condition will still be fulfilled since it is an `||`. I don't think there is a scenario where `hasMatch` is `false` but `matchedInstanceSnap` or `matchedParentMessageRef` are true.